### PR TITLE
Improve the feature about checking update and auto update

### LINKED ISSUE #417: Add "Check updates" button in the app / 更新確認のボタンをつける
## 問題点
APMの更新を確認するために、わざわざこのページを訪れるのは手間がかかります。

## 提案する解決方法
更新確認のボタンをつけていただきたいです。

## 検討した代替案
自動で更新の確認をすることもできるかとは思うのですが、煙たがられることも多いようなので。

### PR DESCRIPTION
**Pull request by hal-shu-sato**
## Description

- Add notification of the update.
- Add options to disable auto-update, notify when an update is available, or download automatically.
- Change the default setting in the zip version to notification only.
- Disable the automatic download option in the zip version.
- Add the button to check an update.

## Related Issue

close #417 

## Motivation and Context

Refer to #417 

## How Has This Been Tested?

- Downgrade `version` in `package.json`.
- Launch apm.
- apm would behave as specified in the configuration.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/renderer/main/preload.ts [hunk 1]
```diff
@@ -57,6 +57,20 @@ window.addEventListener('DOMContentLoaded', async () => {
...
+  const autoUpdateRadios = document.getElementsByName('auto-update');
+  autoUpdateRadios.forEach((element: HTMLInputElement) => {
+    if (element.value === doAutoUpdate) {
+      element.checked = true;
+    }
+  });
+  if (!(await ipcRenderer.invoke('is-exe-version'))) {
+    const e = document.getElementById(
+      'auto-update-download'
+    ) as HTMLInputElement;
```
Comment (yumetodo) [inline src/renderer/main/preload.ts:70]: 一応HTMLInputElementであることを確認するほうが安全には倒せるかもしれないですね・・・？

**Comment by hal-shu-sato** [comment]
I fixed it.

**Review by mitosagi — DISMISSED** [review]
Update works without problems

File: src/main.ts [hunk 2]
```diff
@@ -41,15 +49,117 @@ log.catchErrors({
...
+          });
+        }
+      });
+      request.end();
+    });
+  }
+}
+
+const doAutoUpdate = store.get('autoUpdate') as string;
+if (!isDevEnv) {
```
Comment (yumetodo) [inline src/main.ts:156]:
```suggestion
const doAutoUpdate = store.get('autoUpdate');
if (!isDevEnv && typeof doAutoUpdate  === 'string') {
```

File: src/renderer/main/preload.ts (hunk 1 above)
Comment (yumetodo) [reply src/renderer/main/preload.ts:71]:
```suggestion
    const e = document.getElementById(
      'auto-update-download'
    );
```

4a7777279b97f9365ee106c7b5b6ec1c93d56085 makes you to allow removing `as`.

**Comment by hal-shu-sato** [comment]
I fixed them.

**Review by yumetodo — APPROVED** [review]

**Pull request merged**